Add related and details.rst to 'redundant-unittest-assert' doc (#6483)
PyCQA/pylint: doc/data/messages/a/assert-on-string-literal/details.rst

Directly asserting a string literal will always pass. The solution is to
test something that could fail, or not assert at all.

> This message is similar to ``redundant-unittest-assert``.

Reviewer: ```suggestion
For ``unittest`` assertions there is the similar :ref:`redundant-unittest-assert` message.
```
